app/backend/src/couchers/servicers/admin.py

```diff
@@ -531,3 +532,17 @@ def ListUserIds(self, request, context, session):
             user_ids=user_ids[:page_size],
             next_page_token=str(user_ids[-1]) if len(user_ids) > page_size else None,
         )
+
+    def EditReferenceText(self, request, context, session):
+        reference = session.execute(select(Reference).where(Reference.id == request.reference_id)).scalar_one_or_none()
+
+        if reference is None:
+            context.abort(grpc.StatusCode.NOT_FOUND, errors.REFERENCE_NOT_FOUND)
+
+        if request.new_text.strip() == "":
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.REFERENCE_NO_TEXT)
+
+        reference.text = request.new_text
+        session.commit()
```

Comment:
No need to `session.commit`, the whole request is wrapped in a `session_scope()` context manager that will auto-commit at exit. See here:

https://github.com/Couchers-org/couchers/blob/db5ca2e55d34bfcf20f16be88aeebf95a9a9fe02/app/backend/src/couchers/interceptors.py#L289-L307

https://github.com/Couchers-org/couchers/blob/db5ca2e55d34bfcf20f16be88aeebf95a9a9fe02/app/backend/src/couchers/db.py#L81-L82